images/third_party_bucket.tf

```diff
@@ -0,0 +1,23 @@
+# ------------------------------------------------------------------------------
+# Create an S3 bucket to store third-party files, such as installation
+# packages that are not directly downloadable.
+# ------------------------------------------------------------------------------
+
+resource "aws_s3_bucket" "third_party" {
+  acl    = "private"
+  bucket = format("%s-%s", var.third_party_bucket_name_prefix, lower(local.this_account_type))
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+
+  tags = var.tags
+
+  versioning {
```

Comment:
Do you want versioning here?  This will keep old versions of files if you upload something with the same name.  We want it for the TF state, for example, as an insurance policy.  Normally it's not needed or wanted, though.

Reply:
I'm ok with this not having versioning as well.

Reply:
After a brief team discussion, we agreed that it's fine to leave versioning on, since most files in this bucket will never have more than one version.